### ISSUE TITLE
Syntax highlighting completely broken by string literal type 

### ISSUE THREAD

**Issue by benny-medflyt** [microsoft/TypeScript-TmLanguage#778]
The following code compiles correctly with no errors or warnings, on all TypeScript strictness levels

![ts_syntax_bug_001](https://user-images.githubusercontent.com/24827547/66381385-e3201000-e9c1-11e9-82fa-be0e9a393077.png)

**Code**

```ts
function f<T>(arg: T | null) {
    console.log(arg);
}

class Widget<T, S> {
    constructor(a: T, b: S) {
        console.log(a, b);
    }
}

export function broken() {
    const x = f<{
        x: Widget<string, "x">
    }>(null);
    
    console.log(x);
}

export function foo() {
    console.log();
}

export function bar() {
    console.log();
}
```

**Comment by sheetalkamat** [comment]
This is due to limitation of textmate grammar syntax that doesn't allow multiline lookups. 
Duplicate of #475 and  #773

**Comment by sheetalkamat** [comment]
Fixed by 71c4e0ca325685b72c62daabf27552dd7ea539b0
